vm/systemSmartContracts/validator.go

```diff
@@ -1144,9 +1144,11 @@ func (v *validatorSC) unBond(args *vmcommon.ContractCallInput) vmcommon.ReturnCo
 
 	validatorConfig := v.getConfig(v.eei.BlockChainHook().CurrentEpoch())
 	totalUnBond := big.NewInt(0).Mul(validatorConfig.NodePrice, big.NewInt(int64(len(unBondedKeys))))
-	totalUnBond, returnCode = v.unBondTokensFromRegistrationData(registrationData, totalUnBond)
-	if returnCode != vmcommon.Ok {
-		return returnCode
+	if len(unBondedKeys) > 0 {
+		totalUnBond, returnCode = v.unBondTokensFromRegistrationData(registrationData, totalUnBond)
```

Comment:
what happened if we have returned here directly?

Reply:
Maybe some changes in save will not happen